tests/steps/application_v2_steps.py

```diff
@@ -1132,16 +1136,28 @@ def check_found_method_or_error(context, methodsig, error: str = None):
 
 
 @then(
-    'the contents of the box with name "{box_name}" in the current application should be "{box_value:MaybeString}". If there is an error it is "{error_string:MaybeString}".'
+    'according to "{from_client}", the contents of the box with name "{box_name}" in the current application should be "{box_value:MaybeString}". If there is an error it is "{error_string:MaybeString}".'
 )
 def check_box_contents(
-    context, box_name, box_value: str = None, error_string: str = None
+    context,
+    from_client,
+    box_name,
+    box_value: str = None,
+    error_string: str = None,
 ):
     try:
         box_name = split_and_process_app_args(box_name)[0]
-        box_response = context.app_acl.application_box_by_name(
-            context.current_application_id, box_name
-        )
+        if from_client == "algod":
+            box_response = context.app_acl.application_box_by_name(
+                context.current_application_id, box_name
+            )
+        elif from_client == "indexer":
+            box_response = context.app_icl.application_box_by_name(
+                context.current_application_id, box_name
+            )
+        else:
+            assert False, "expecting algod or indexer, get " + from_client
```

Comment:
nit to use f-strs :
```suggestion
            assert False, f"expecting algod or indexer, got: {from_client}"
```

Reply:
oh boy, I don't know that we can actually use f-strs.